x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Driver.java

```diff
@@ -354,4 +375,22 @@ private DriverStatus updateStatus(DriverStatus.Status status) {
             activeOperators.stream().map(o -> new DriverStatus.OperatorStatus(o.toString(), o.status())).toList()
         );
     }
+
+    private static class StoredThreadContext {
```

Comment:
the usage of this looks similar to `org.elasticsearch.action.support.ContextPreservingActionListener` , could that work here instead ? Dealing with thread context is tricky and the try-with-resource provides some nice guarantees. (the code is not obvious but when the try with resources in ContextPreservingActionListener terminates will restore the passed in context)

Reply:
@jakelandis I tried using the ContextPreservingActionListener before, but it didn't work as expected. I believe the issue was that I reserved some context twice. I've made this work with 0bf6c22. Thanks for pushing on this :).